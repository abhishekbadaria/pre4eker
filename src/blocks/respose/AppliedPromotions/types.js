const types = {
  earnedBonusPoints: "Начисление баллов",
  spentBonusPoints: "Скидка балламми",
  deliveryDiscount: "Скидка на доставку",
  message: "Сообщение",
  issuedCoupon: "Выданный промокод",
  discount: "Скидка",
};

export default types;