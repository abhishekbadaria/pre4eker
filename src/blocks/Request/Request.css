.request {
  background: #FFFFFF;
  box-shadow: 0px 1px 4px rgba(0, 0, 0, 0.15);
  /* width: 35%; */
  padding: 16px 26px;
}

.h2 {
  font-size: 18px;
  line-height: 22px;
}