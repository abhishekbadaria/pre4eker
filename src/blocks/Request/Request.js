import React, { Component } from 'react'

import { Input } from "./../Input/Input";
import style from './Request.css';
import Selector from '../Selector/Selector'

export default class Request extends Component {
  constructor(props) {
    super(props)
    
  }
  


  render() {
    const customerSelectedMethod = this.props.customer.selectorReducer.selected
    const customerSelectedField = this.props.customer.selectorReducer.options[customerSelectedMethod];
  console.log(customerSelectedField) 
    return (
      <div className={style.request}>
        <h2 className={style.h2}>Клиент</h2>
        <Selector field={customerSelectedField} options={this.props.customer.selectorReducer.options} changeSelection={this.props.changeSelection} />
      </div>
    )
  }
}
