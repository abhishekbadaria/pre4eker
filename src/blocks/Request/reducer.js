import request from './requestObj';

export const requestReducer = function(state = request, action) {
  // console.log(state)
  switch (action.type) {
    case "SELECT_CUSTOMER":
      const {customer} = state;
      customer.selected = action.selected;
      return {...state, customer}; 
    case "SELECT_ORDER_IDENTIFIER":
      const {order} = state;
      order.identifyer.selected = action.selected;
      return {...state, order}; 
    case "ADD_CUSTOM_FIELD":
      const {customfields} = state.order;
      customfields.fields.push(1);
      return {...state, customfields}; 
    default:
      return state;
  }
};
