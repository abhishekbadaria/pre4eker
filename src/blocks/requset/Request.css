.buttonLine {
  display: flex;
  margin-bottom: .5rem;
}

.buttonLine button {
  margin-right: .2rem;
}
.buttonLine button:last-child {
  margin-right: 0;
}