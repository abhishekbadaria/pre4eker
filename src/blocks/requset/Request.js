import React, { Component } from "react";
import PropTypes from "prop-types";

import { RequestFormContainer } from "./RequestForm/RequestFormContainer";
import InputJSON from "./InputJSON/InputJSON";
import style from "./Request.css";
import Button from "src/blocks/Button/Button";
import Soon from "./Soon/Soon";
import * as formTypes from "./formTypes";

const Request = (props) => {
  const {
    changeFormType,
    requestFromType,
    setRequestJSON,
    setResponseJSON,
  } = props;

  const drawRequesForm = (type) => {
    switch (type) {
      case formTypes.requestForm.type:
        return (
          <div className={style.requestForm}>
            <Soon />
          </div>
        );
      case formTypes.reqiestJSON.type:
        return (
          <div className={style.requestForm}>
            <Soon />
          </div>
        );
      case formTypes.responseJSON.type:
        return (
          <div className={style.requestJSON}>
            <InputJSON setJSON={setResponseJSON} />
          </div>
        );

      default:
        break;
    }
  };

  const buttons = [];

  return (
    <div>
      <div className={style.buttonLine}>
        {Object.keys(formTypes).map((button, i) => (
          <Button
            key={`button_${i}`}
            type="TEXT"
            action={() => changeFormType(formTypes[button].type)}
            size="sizeAuto"
            active={requestFromType.type === formTypes[button].type}
          >
            {formTypes[button].text}
          </Button>
        ))}
      </div>
      {drawRequesForm(requestFromType.type)}
    </div>
  );
};

Request.propTypes = {};

export default Request;
