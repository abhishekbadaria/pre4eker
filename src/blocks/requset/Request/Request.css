.request {
  background: #FFFFFF;
  box-shadow: 0px 1px 4px rgba(0, 0, 0, 0.15);
  width: 400px;
  padding: 16px 26px;
}

.h2 {
  font-size: 18px;
  line-height: 22px;
}