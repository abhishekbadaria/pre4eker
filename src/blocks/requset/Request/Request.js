import React, { Component } from "react";
 
import style from "./Request.css";
import Customer from '../Customer/Customer';
import Order from '../Order/Order';


export default class Request extends Component {
  constructor(props) {
    super(props);
    this.customer = this.props.request.requestReducer.customer;
    this.order = this.props.request.requestReducer.order
  }

  render() {
 

    return (
      <div className={style.request}>
        <Customer customer={this.customer}/>
        <Order  order={this.order}/>
        

      </div>
    );
  }
}
