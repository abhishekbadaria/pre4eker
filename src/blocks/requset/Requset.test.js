import React from "react";
import Request from "./Request";
import {
  render,
  fireEvent,
  waitFor,
  screen,
  cleanup,
  act,
} from "@testing-library/react";

import { exportAllDeclaration } from "@babel/types";

afterEach(cleanup);

const defaultRequestForm = {
  requstFormType: `RESPONSE_INPUT`,
  changeFormType: jest.fn(),
};

test(`customer match snapshot`, () => {
  const { asFragment, getByRole } = render(
    <Request requestForm={defaultRequestForm} />
  );
  expect(asFragment(<Request />)).toMatchSnapshot();
});
