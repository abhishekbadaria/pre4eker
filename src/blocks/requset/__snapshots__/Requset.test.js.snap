// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`customer match snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="buttonLine"
    >
      <button>
        Форма запроса
      </button>
      <button>
        JSON запроса
      </button>
      <button>
        JSON ответа
      </button>
    </div>
    <div />
  </div>
</DocumentFragment>
`;
